### PR TITLE
add sendchatevent to 3.0

### PR DESCRIPTION
**Pull request by BlazeMCworld**
simply removes the old testevent and adds a new sendchatevent with the message field that runs whenever the player types in chat, previously was being done directly in the mlocalplayer mixin

### CODE REVIEW

File: common/src/main/java/io/github/codeutilities/event/impl/SendChatEvent.java [hunk 1]
```diff
@@ -2,6 +2,6 @@
 
 import io.github.codeutilities.event.Event;
 
-public class TestEvent implements Event {
+public record SendChatEvent(String message) implements Event {
```
Comment (Reasonlesss) [inline common/src/main/java/io/github/codeutilities/event/impl/SendChatEvent.java]: make this a cancelable event

**Review by Reasonlesss — CHANGES_REQUESTED** [review]

File: common/src/main/java/io/github/codeutilities/event/type/Cancellable.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package io.github.codeutilities.event.type;
+
+public class Cancellable {
```
Comment (Reasonlesss) [inline common/src/main/java/io/github/codeutilities/event/type/Cancellable.java]: a class can only be extended once, we won't be able to have different stuff in the future

File: common/src/main/java/io/github/codeutilities/mixin/MLocalPlayer.java [hunk 3]
```diff
@@ -23,7 +23,11 @@ private void chat(String msg, CallbackInfo ci) {
                     ci.cancel();
                 }
             }
-            EventHandler.invoke(new SendChatEvent(msg));
+            SendChatEvent evn = new SendChatEvent(msg);
```
Comment (Reasonlesss) [inline common/src/main/java/io/github/codeutilities/mixin/MLocalPlayer.java:26]: could just be named `event` for readability

File: common/src/main/java/io/github/codeutilities/scripts/action/ScriptActionType.java [hunk 4]
```diff
@@ -1,156 +1,164 @@
 package io.github.codeutilities.scripts.action;
 
 import io.github.codeutilities.CodeUtilities;
+import io.github.codeutilities.event.Event;
+import io.github.codeutilities.event.type.Cancellable;
 import io.github.codeutilities.util.ComponentUtil;
-import java.util.function.BiConsumer;
+import org.apache.logging.log4j.util.TriConsumer;
```
Comment (Reasonlesss) [inline common/src/main/java/io/github/codeutilities/scripts/action/ScriptActionType.java]: log4j?

**Review by Reasonlesss — CHANGES_REQUESTED** [review]

File: common/src/main/java/io/github/codeutilities/event/type/Cancellable.java (hunk 2 above)
Comment (BlazeMCworld) [reply common/src/main/java/io/github/codeutilities/event/type/Cancellable.java]: image
wat, works for me

File: common/src/main/java/io/github/codeutilities/scripts/action/ScriptActionType.java (hunk 4 above)
Comment (BlazeMCworld) [reply common/src/main/java/io/github/codeutilities/scripts/action/ScriptActionType.java]: wat, didnt even see that, should i create a new class that contains all 3 properties and then use a normal consumer?

File: common/src/main/java/io/github/codeutilities/event/type/Cancellable.java (hunk 2 above)
Comment (BlazeMCworld) [reply common/src/main/java/io/github/codeutilities/event/type/Cancellable.java]: nvm me, get what you mean, but how would you recommend fixing it

Comment (Reasonlesss) [reply common/src/main/java/io/github/codeutilities/event/type/Cancellable.java]: an interface ig

Comment (BlazeMCworld) [reply common/src/main/java/io/github/codeutilities/event/type/Cancellable.java]: tried, interfaces cant have modifiable fields, unless i did something wrong

File: common/src/main/java/io/github/codeutilities/scripts/action/ScriptActionType.java (hunk 4 above)
Comment (Reasonlesss) [reply common/src/main/java/io/github/codeutilities/scripts/action/ScriptActionType.java]: probably just our own

Comment (BlazeMCworld) [reply common/src/main/java/io/github/codeutilities/scripts/action/ScriptActionType.java]: so own triconsumer or own class with 3 properties?

Comment (Reasonlesss) [reply common/src/main/java/io/github/codeutilities/scripts/action/ScriptActionType.java]: ye

File: common/src/main/java/io/github/codeutilities/event/type/Cancellable.java (hunk 2 above)
Comment (Reasonlesss) [reply common/src/main/java/io/github/codeutilities/event/type/Cancellable.java]: yeah they can't thats just java so we're probably gonna have to just scrap this change

**Review by Reasonlesss — APPROVED** [review]

File: common/src/main/java/io/github/codeutilities/mixin/MLocalPlayer.java (hunk 3 above)
Comment (Reasonlesss) [reply common/src/main/java/io/github/codeutilities/mixin/MLocalPlayer.java:26]: change this next pr

**Pull request merged**